src/main/starlark/builtins_bzl/common/cc/cc_binary.bzl

```diff
@@ -594,6 +594,19 @@ def cc_binary_impl(ctx, additional_linkopts, force_linkstatic = False):
         pdb_file = ctx.actions.declare_file(_strip_extension(binary) + ".pdb", sibling = binary)
         additional_linker_outputs.append(pdb_file)
 
+    # On macOS, if GENERATE_DSYM_FILE feature is enabled
+    # then a .dSYM file will be built along with the executable.
+    dsym_file = None
+    if cc_common.is_enabled(feature_configuration = feature_configuration, feature_name = "generate_dsym_file"):
+        dsym_file = ctx.actions.declare_directory(
```

Comment:
I think this needs to be a file.  Historically the toolchain calls dsymutil with --flat to generate a flat file, and that still seems to true for apple_support today.  I suspect there might be a desire to support dsymutil generating a directory also, but let's not worry about that for this PR.